hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSUtil.java

```diff
@@ -1295,6 +1296,22 @@ static URI trimUri(URI uri) {
    */
   public static void addPBProtocol(Configuration conf, Class<?> protocol,
       BlockingService service, RPC.Server server) throws IOException {
+    RPC.setProtocolEngine(conf, protocol, ProtobufRpcEngine2.class);
+    server.addProtocol(RPC.RpcKind.RPC_PROTOCOL_BUFFER, protocol, service);
+  }
+
+  /**
+   * Add protobuf based protocol to the {@link org.apache.hadoop.ipc.RPC.Server}
+   * @param conf configuration
+   * @param protocol Protocol interface
+   * @param service service that implements the protocol
+   * @param server RPC server to which the protocol &amp; implementation is
+   *               added to
+   * @throws IOException
+   */
+  @Deprecated
+  public static void addPBProtocol(Configuration conf, Class<?> protocol,
```

Comment:
Memo: The method is used from ozone.
```
➜  hadoop-ozone git:(master) find . -name "*.java" | xargs grep "addPBProtocol"
./hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:    DFSUtil.addPBProtocol(conf, protocol, instance, rpcServer);
./hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/SCMTestUtils.java:    DFSUtil.addPBProtocol(conf, protocol, instance, rpcServer);
./hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:    DFSUtil.addPBProtocol(conf, protocol, instance, rpcServer);
```

Reply:
Filed https://issues.apache.org/jira/browse/HDDS-3678 for Ozone.

Reply:
https://github.com/apache/hadoop-ozone/pull/987 is now available. If the PR is merged, I think we don't have to fix DFSUtil.java.

Reply:
But, I feel, we can still keep the method with deprecated.